hasher-matcher-actioner/hmalib/common/config.py

```diff
@@ -126,13 +136,182 @@ def initialize(config_table_name: str) -> None:
         _TABLE_NAME = config_table_name
 
 
+class _HMAConfigWithSubtypeMeta(type):
+    """
+    Metaclass to connect subtypes and types, provide some defaults
+    """
+
+    def __new__(metacls, cls_name: str, bases, cls_dict):
+        # Is this one of the bases?
+        if cls_name in ("HMAConfigWithSubtypes", "_HMAConfigSubtype"):
+            return super().__new__(metacls, cls_name, bases, cls_dict)
+        # Has a Subtype already been applied?
+        for base in bases:
+            if hasattr(base, "Subtype"):
+                return super().__new__(metacls, cls_name, bases, cls_dict)
+        # Else create magic defaults
+        cls_dict["Subtype"] = None  # Recursion guard, overwrite below
+
+        config_type = ""
+        # Provide magic default - last in MRO order or class name
+        for base in bases:
+            if hasattr(base, "CONFIG_TYPE"):
+                config_type = base.CONFIG_TYPE
+        if not config_type:
+            config_type = cls_name
+            cls_dict["CONFIG_TYPE"] = cls_name
```

Comment:
These lines should not be needed, because SubType and CONFIG_TYPE have the same lifetimes, and we exit early above, this will just duplicate CONFIG_TYPE on subclasses